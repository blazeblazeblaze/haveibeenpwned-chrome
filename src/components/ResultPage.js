import React from 'react';
import NotPwned from './NotPwned.js';
import Pwned from './Pwned.js';
import PropTypes from 'prop-types';

const ResultPage = ({isPwned, onStartOver, email}) => {
  return(
    <div>
      {isPwned && <Pwned onStartOver={onStartOver} email={email} />}
      {!isPwned && <NotPwned onStartOver={onStartOver} />}
    </div>
  );
}

ResultPage.propTypes = {
  isPwned: PropTypes.bool.isRequired,
  onStartOver: PropTypes.func.isRequired,
  email: PropTypes.string
};

ResultPage.defaultProps = {
  email: ''
};

export default ResultPage;
