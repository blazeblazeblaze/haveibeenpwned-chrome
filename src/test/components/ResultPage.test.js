import React from 'react';
import ReactDOM from 'react-dom';
import ResultPage from '../../components/ResultPage';
import renderer from 'react-test-renderer';

jest.mock('../../components/Pwned', () => () => 'PwnedComponent');
jest.mock('../../components/NotPwned', () => () => 'NotPwnedComponent');

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ResultPage onStartOver={jest.fn } isPwned={false} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('ResultPage renders Pwn when isPwned is true', () => {
  const component = renderer.create(
    <ResultPage onStartOver={jest.fn} isPwned={true} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ResultPage renders NonPwn when isPwned is false', () => {
  const component = renderer.create(
    <ResultPage onStartOver={jest.fn} isPwned={false} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
