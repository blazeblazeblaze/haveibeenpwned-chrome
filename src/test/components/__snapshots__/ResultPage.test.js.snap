// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPage renders NonPwn when isPwned is false 1`] = `
<div>
  NotPwnedComponent
</div>
`;

exports[`ResultPage renders Pwn when isPwned is true 1`] = `
<div>
  PwnedComponent
</div>
`;
